client/src/pages/public/Otp.js

```diff
@@ -0,0 +1,111 @@
+import { Page } from '../../components/generic';
+import React, { useState } from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import { Link } from 'react-router-dom';
+import ContactMailIcon from '@material-ui/icons/ContactMail';
+import * as qs from 'querystring';
+import InfoIcon from '@material-ui/icons/Info';
+import { Typography } from '@material-ui/core';
+import { Button } from '../../components/generic/Button';
+import { Card } from '../../components/generic/Card';
+import Box from '@material-ui/core/Box';
+import CheckBoxIcon from '@material-ui/icons/CheckBox';
+
+const useStyles = makeStyles((theme) => {
+  return {
+    root: {
+      marginTop: 100,
+      width: 376,
+      height: 270,
+    },
+    title: {
+      paddingBottom: 10,
+    },
+    button: {
+      paddingTop: 10,
+      justifySelf: 'flex-end',
+      alignItems: 'flex-end',
+    },
+    icon: {
+      alignSelf: 'center',
+      paddingBottom: 10,
+      color: theme.palette.primary.dark,
+    },
+    check: {
+      alignSelf: 'center',
+      paddingBottom: 10,
+      color: 'green',
+    },
+    error: {
+      alignSelf: 'center',
+      paddingBottom: 10,
+      color: 'red',
+    },
+  };
+});
+
+export default (props) => {
+  const [pageState, setPageState] = useState('default');
+  const classes = useStyles();
+  // Get the query string from the url
+  // Remove the questionmark from the search
+  const query = qs.parse(props.location.search.slice(1), '&', '=');
```

Comment:
Is importing location as a prop necessary here? I've used window.location in the past.

Reply:
I played around with both. `window.location` gave me the full url, where props.location gave me a more detailed object with just the `search` as one of the options.  

I felt this was cleaner.

Reply:
Cool, I'm on board with clean code!

Reply:
Just to add to the conversation, I'd vouch for react router + native classes like in this example, not necessary to change anything just tossing in my 2c :p

https://reactrouter.com/web/example/query-parameters